.github/workflows/Tests.yaml

```diff
@@ -224,6 +224,7 @@ jobs:
     name: Unit tests
     runs-on: ubuntu-latest
     strategy:
+      fail-fast: false
```

Comment:
Before this change, if the tests fail for any of the compiler options (for example "gcc-7 Debug ON") then the tests are not even run for the other options.

After this change, all the tests are run, even if some of them fail.

Not sure how important this is, but I wanted it for debugging because it wasn't clear to me why the tests were failing for some compilers but succeeding for others, and when the first compiler that was tried was a failure, I didn't even know that the tests succeeded for other compilers.

Reply:
Please make sure to drop this change before merging it into `develop`, perhaps prefix the commit message with `DROP` to remember

Reply:
At one of the SpECTRE meetings, it was suggested that this change be made permanent because we are not limited by the speed of failing tests.  So that's why I included it in this PR.

Reply:
In that case, please make it a separate PR, so it can get the attention it deserves :)

Reply:
Ok, done. #2407